lib/ini.js

```diff
@@ -104,16 +123,29 @@ function splitSections (str, separator) {
 }
 
 const decode = (str, opt = {}) => {
+  // The `typeof` check is required because accessing the `process` directly fails on browsers.
+  /* istanbul ignore next */
+  opt.platform = opt.platform || (typeof process !== 'undefined' && process.platform)
+  /* istanbul ignore next */
+  const eol = opt.platform === 'win32' ? '\r\n' : '\n'
   opt.bracketedArray = opt.bracketedArray !== false
   const out = Object.create(null)
   let p = out
   let section = null
+  const preserveComments = opt.preserveComments || false
+  const commentsKey = opt.commentsKey || 'comments'
+  const commentsDictionary = {}
+  let lineCommentArray = []
+  const commentsRegEx = /^[#;]/
   //          section          |key      = value
   const re = /^\[([^\]]*)\]\s*$|^([^=]+)(=(.*))?$/i
   const lines = str.split(/[\r\n]+/g)
   const duplicates = {}
 
   for (const line of lines) {
+    if (line && line.match(commentsRegEx)) {
+      lineCommentArray.push(line)
+    }
     if (!line || line.match(/^\s*[;#]/) || line.match(/^\s*$/)) {
```

Comment:
This line has the regex inline, so the `commentsRegex` can probably be an inline one also.  That's one less line of thought a person has to do when reading the code.

Reply:
Do you need me to clean all these commits up and submit a new cleaner PR?

Reply:
No, we squash when we merge so as long as things are heading in the right direction you can just keep pushing changes to your branch.

I know this new PR was based off of suggestions from @lukekarrys so I'll defer to them if this is going in the right direction generally.